tfe/data_source_github_app_installation_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tfe
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"testing"
+)
+
+func testAccTFEGHAInstallationDataSourcePreCheck(t *testing.T) {
+	testAccPreCheck(t)
+	if envGithubAppInstallationID == "" {
+		t.Skip("Please set GITHUB_APP_INSTALLATION_ID to run this test")
+	}
+}
+
+// For manual testing a GitHub app installation name has to be provided
+func TestAccTFEGHAInstallationDataSource_findByName(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccTFEGHAInstallationDataSourcePreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEGHAInstallationDataSourceConfig_findByName("installation_name"),
```

Comment:
Is this installation name another kind of test dependency? Does the name of the installation need to be "installation_name" ?

Reply:
@brandonc yes, for this data source to work, either installation_name or id needs to be passed as an argument.

Reply:
Right, but my question is about whether this particular name needs to be part of the ENV test setup because it appears that an installation named "installation_name" is required for this test to pass.

Reply:
Yes, I can add it to the Env test setup, but its needed just for this one test. Also the automation test for this is being skipped (based on GITHUB_INSTALLATION_ID check), and its only the manual testing that would need it. So I didn't add it to the ENV test setup and left it to be hardcoded during the manual testing.

Reply:
I think it needs to be extracted out to ENV alongside `GITHUB_INSTALLATION_ID` in the docs... Maybe call it `GITHUB_INSTALLATION_NAME` and default it to "installation_name" if you want to... but the dependency is real and needs to be accounted for if we ever want to run these tests again.

Reply:
Done.